src/main/java/organice/model/person/Nric.java

```diff
@@ -32,10 +59,47 @@ public Nric(String nric) {
     }
 
     /**
-     * Returns true if a given string is a valid nric.
+     * Returns true if a given string is a valid Nric
      */
     public static boolean isValidNric(String test) {
-        return test.toUpperCase().matches(VALIDATION_REGEX);
+        requireNonNull(test);
+        if (!test.toUpperCase().matches(VALIDATION_REGEX)) {
+            return false;
+        }
+
+        return test.toUpperCase().charAt(test.length() - 1) == calculateChecksumLetter(test);
```

Comment:
tbh i prefer a method that give me boolean `isValidChecksum` that takes in nric and return true or false, so you can just use `test = test.toUpperCase() -> return test.matches(VALIDATION_REGEX) && isValidChecksum(test)`? What do you think

Reply:
I would agree but I am also using this method to generate the checksums for `SampleDataUtil`